webbrowser/open_browser_unix.go

```diff
@@ -9,27 +9,35 @@ import (
 	"os/exec"
 )
 
-// OpenInBrowser checks for the presence of a display server
-// and environment variables indicating a gui is present. If found
-// then xdg-open is called on a url to open said url in the default
-// gui web browser for the system
+// Open opens `url` in default system browser. It tries to do so in two
+// ways (xdg-open and $BROWSER). It only works if there is a display
+// server working.
+//
+// bouncepaw: I tried to support TTYs as well. The idea was to open
+// a browser in foreground and return back to amfora after the browser
+// is closed. While all browsers I tested opened correctly (w3m, lynx),
+// I couldn't make it restore amfora correctly. The screen always ended
+// up distorted. None of my stunts with altscreen buffers helped.
 func Open(url string) (string, error) {
-	disp := os.Getenv("DISPLAY")
-	wayland := os.Getenv("WAYLAND_DISPLAY")
-	_, err := exec.LookPath("Xorg")
-	if disp == "" && wayland == "" && err != nil {
+	var (
+		// In prev versions there was also Xorg executable checked for.
+		// I don't see any reason to check for it.
+		xorgDisplay                     = os.Getenv("DISPLAY")
+		waylandDisplay                  = os.Getenv("WAYLAND_DISPLAY")
+		xdgOpenPath, xdgOpenNotFoundErr = exec.LookPath("xdg-open")
+		envBrowser                      = os.Getenv("BROWSER")
+	)
+	switch {
+	case xorgDisplay == "" && waylandDisplay == "":
 		return "", fmt.Errorf("no gui is available")
+	case xdgOpenNotFoundErr == nil || envBrowser != "":
```

Comment:
Yeah, that's a mistake. There were two similar `case`-blocks and I smashed them together without second thought. Gotta fix this.

I think `xdg-open` should be used first because it's more idiomatic (I think) and is also similar to how this function is implemented for other platforms.

Reply:
I thought `$BROWSER` should be first because it's something the user manually sets, so it's more likely to be a custom specific choice than `xdg-open`. Plus it's only used by CLI software, so a user could use it to differentiate between links in CLI and GUI.

Reply:
That's a good point (you've almost convinced me), but consider `xdg-open`'s [source code](https://github.com/freedesktop/xdg-utils/blob/master/scripts/xdg-open.in#L434). It seems to me that `xdg-open` depends on the `$BROWSER` variable in some cases. Therefore, `xdg-open` is more universal and should be used first.

Reply:
Oh, I had no idea! Ok, I agree. I will merge this soon, consider it complete.